build_rules.gradle

```diff
@@ -1253,6 +1254,125 @@ ext.applyGrpcNature = {
 
 /*************************************************************************************************/
 
+ext.applyPortabilityNature = {
+  println "applyPortabilityNature with " + (it ? "$it" : "default configuration") + " for project $project.name"
+  applyJavaNature(enableFindbugs: false, enableErrorProne: false, shadowClosure: {
+    // guava uses the com.google.common and com.google.thirdparty package namespaces
+    relocate "com.google.common", "org.apache.beam.vendor.guava.v20.com.google.common"
+    relocate "com.google.thirdparty", "org.apache.beam.vendor.guava.v20.com.google.thirdparty"
+
+    relocate "com.google.protobuf", "org.apache.beam.vendor.protobuf.v3.com.google.protobuf"
```

Comment:
Is there a way to error out if a jar contains a package that is not relocated here ?

Reply:
That is already being done by the `validateShadedJarDoesntLeakNonOrgApacheBeamClasses` task.